packages/bruno-app/src/components/Environments/EnvironmentSettings/EnvironmentList/index.js

```diff
@@ -46,46 +62,109 @@ const EnvironmentList = ({ collection }) => {
     }
   }, [envUids, environments, prevEnvUids]);
 
+  // Prevent switching to another environment if isModified is true
+  const handleEnvironmentClick = (env) => {
+    if (!isModified) {
```

Comment:
`isModified` is derived from `formik.dirty` I think you can just use `formik.dirty` here and the other places.